### PR TITLE
Enable build_gallery tests in try/prod builders

### PR DESCRIPTION
**Pull request by keyonghan**
## Description

This PR enabled build_gallery builders (the old deploy_gallery tests in cirrus) for pre-submit tests.

## Related Issues

https://github.com/flutter/flutter/issues/66707

### CODE REVIEW

File: dev/prod_builders.json [hunk 1]
```diff
@@ -24,6 +24,12 @@
          "task_name": "linux_docs",
          "flaky": false
       },
+      {
```
Comment (godofredoc) [inline dev/prod_builders.json:27]: Can we also remove them from cirrus?

Comment (keyonghan) [reply dev/prod_builders.json:27]: Removed.

File: .cirrus.yml [hunk 2]
```diff
@@ -240,16 +229,6 @@ task:
...
-      # even if a test has failed. The behavior of failing dependencies is non-ideal for infra health.
-      # See also: https://github.com/flutter/flutter/pull/49454
-      depends_on:
-        - analyze-linux
-      script:
-        - ulimit -S -n 2048 # https://github.com/flutter/flutter/issues/2976
-        - ./dev/bots/deploy_gallery.sh
-
     - name: verify_binaries_codesigned-macos # macos-only
       # TODO(fujino): remove this `only_if` after https://github.com/flutter/flutter/issues/44372
```
Comment (godofredoc) [inline .cirrus.yml:233]: We can also delete this one.

Comment (keyonghan) [reply .cirrus.yml]: Removed the entire mac section.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**